homeassistant/components/recorder/__init__.py

```diff
@@ -266,13 +270,23 @@ def do_purge(now):
                     self.queue.task_done()
                     continue
 
-            with session_scope(session=self.get_session()) as session:
+            dbevent = None
```

Comment:
This is still way too difficult. Just add this ` elif event.event_type == EVENT_TIME_CHANGED:`:

```python
if event.event_type in self.exclude_t:
    self.queue.task_done()
    continue
```